analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/ExecutingSqlQueries.cs

```diff
@@ -101,29 +88,38 @@ expression switch
+            semanticModel.GetDeclaringSyntaxNode(identifier) is VariableDeclaratorSyntax variableDeclaratorSyntax
+            && IsInterpolated(variableDeclaratorSyntax.Initializer?.Value, semanticModel);
+
         private static bool IsStringMethodInvocation(InvocationExpressionSyntax invocation, string methodName, SemanticModel semanticModel) =>
-            invocation.IsMethodInvocation(KnownType.System_String, methodName, semanticModel) &&
-            !AllConstants(invocation.ArgumentList.Arguments.ToList(), semanticModel);
+            invocation.IsMethodInvocation(KnownType.System_String, methodName, semanticModel)
+            && !AllConstants(invocation.ArgumentList.Arguments.ToList(), semanticModel);
 
-        private bool IsStringMethodInvocation(IdentifierNameSyntax identifier, string methodName, SemanticModel semanticModel)
-            => semanticModel.GetDeclaringSyntaxNode(identifier) is VariableDeclaratorSyntax variableDeclaratorSyntax &&
-               IsStringMethodInvocation(methodName, variableDeclaratorSyntax.Initializer?.Value, semanticModel);
+        private bool IsStringMethodInvocation(IdentifierNameSyntax identifier, string methodName, SemanticModel semanticModel) =>
+            semanticModel.GetDeclaringSyntaxNode(identifier) is VariableDeclaratorSyntax variableDeclaratorSyntax
+            && IsStringMethodInvocation(methodName, variableDeclaratorSyntax.Initializer?.Value, semanticModel);
 
         private static bool AllConstants(IEnumerable<ArgumentSyntax> arguments, SemanticModel semanticModel) =>
             arguments.All(a => a.Expression.IsConstant(semanticModel));
 
         private static bool IsConcatenationOfConstants(BinaryExpressionSyntax binaryExpression, SemanticModel semanticModel)
         {
-            System.Diagnostics.Debug.Assert(binaryExpression.IsKind(SyntaxKind.AddExpression));
-            if ((semanticModel.GetTypeInfo(binaryExpression).Type is ITypeSymbol concantenationType) &&
-                binaryExpression.Right.IsConstant(semanticModel))
+            System.Diagnostics.Debug.Assert(binaryExpression.IsKind(SyntaxKind.AddExpression), "Binary expression should be of syntax kind add expression.");
+            if ((semanticModel.GetTypeInfo(binaryExpression).Type is ITypeSymbol) && binaryExpression.Right.IsConstant(semanticModel))
             {
                 var nestedLeft = binaryExpression.Left;
                 var nestedBinary = nestedLeft as BinaryExpressionSyntax;
                 while (nestedBinary != null)
                 {
-                    if (!nestedBinary.Right.IsConstant(semanticModel) ||
-                        (!nestedBinary.IsKind(SyntaxKind.AddExpression) && !nestedBinary.IsConstant(semanticModel)))
+                    if (!nestedBinary.Right.IsConstant(semanticModel)
+                        || (!nestedBinary.IsKind(SyntaxKind.AddExpression)
+                            && !nestedBinary.IsConstant(semanticModel)))
```

Comment:
This one is seriously hard to read.
v1: Let's keep the `&&` on single line to see the main `||` logic clearly.
```suggestion
                    if (!nestedBinary.Right.IsConstant(semanticModel)
                        || (!nestedBinary.IsKind(SyntaxKind.AddExpression) && !nestedBinary.IsConstant(semanticModel)))
```
v2: Can we invert the condition? That should make it readable. Something like
```
if(right is constant && (nested is constant || nested is add))
  do the nesting
else
  return false
```

Reply:
Waaay better :) Now we can see what it does